### ISSUE TITLE
Implement Reflection

### ISSUE THREAD

**Issue by 7ap** [7ap/rbx#1]
We need to implement the higher-level functions associated with reading/writing data using Reflection before we reach 0.1.0, in the process we likely will need to create new structures to fill parts I've missed.

**Comment by crossliner** [comment]
the proper way of getting the lua state from scriptcontext

**Comment by CompeyDev** [comment]
A bit off topic, but:

Once this reaches a stable release, would it be logically possible to use the bindings and create an alternative frontend to studio?

**Comment by EpixScripts** [comment]
> A bit off topic, but:
> 
> Once this reaches a stable release, would it be logically possible to use the bindings and create an alternative frontend to studio?

The project relies on the running process of Roblox Studio. Using this alone, you would have to do something like hiding the regular window that Studio creates, or prevent the GUI from being initialized.

**Comment by u-train** [comment]
You would at best have an API to reimplement with all of its quirks, also keep track of an impossible moving target, too…

**Comment by CompeyDev** [comment]
> You would at best have an API to reimplement with all of its quirks, also keep track of an impossible moving target, too…

Yup, that's the problem with trying to recreate its backend, considering how ever evolving it is. Hooking onto the already well maintained backend would be a better strategy instead.

**Comment by CompeyDev** [comment]
> > A bit off topic, but:
> > Once this reaches a stable release, would it be logically possible to use the bindings and create an alternative frontend to studio?
> 
> The project relies on the running process of Roblox Studio. Using this alone, you would have to do something like hiding the regular window that Studio creates, or prevent the GUI from being initialized.

This seems doable, will look into it.

**Comment by crossliner** [comment]
prob it's related but is there any plans for headless roblox studio?
